src/controller/scan/callback.go

```diff
@@ -0,0 +1,110 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scan
+
+import (
+	"context"
+
+	"github.com/goharbor/harbor/src/controller/artifact"
+	"github.com/goharbor/harbor/src/controller/event/metadata"
+	"github.com/goharbor/harbor/src/controller/robot"
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/notification"
+	"github.com/goharbor/harbor/src/pkg/scan"
+	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+const (
+	// ScanAllCallback the scheduler callback name of the scan all
+	ScanAllCallback = "scanAll"
+)
+
+func init() {
+	scheduler.RegisterCallbackFunc(ScanAllCallback, scanAllCallback)
+
+	// NOTE: the vendor type of execution for the scan job trigger by the scan all is job.ImageScanAllJob
+	task.RegisterCheckInProcessor(job.ImageScanAllJob, scanTaskCheckInProcessor)
+
+	task.RegisterCheckInProcessor(job.ImageScanJob, scanTaskCheckInProcessor)
+
+	task.RegisterTaskStatusChangePostFunc(job.ImageScanJob, scanTaskStatusChange)
+}
+
+func scanAllCallback(ctx context.Context, param string) error {
+	_, err := DefaultController.ScanAll(ctx, task.ExecutionTriggerSchedule, true)
+	return err
+}
+
+func scanTaskStatusChange(ctx context.Context, taskID int64, status string) (err error) {
+	logger := log.G(ctx).WithFields(log.Fields{"task_id": taskID, "status": status})
+
+	js := job.Status(status)
+
+	if js.Final() {
+		t, err := task.Mgr.Get(ctx, taskID)
```

Comment:
It seems here if the task manager can inject the task object itself, duplicated code can be reduced, right?

@ywk253100